ssm/agents.py

```diff
@@ -392,3 +396,31 @@ def get_dns(dn_file, log):
 
     log.debug('%s DNs found.', len(dns))
     return dns
+
+
+def get_banned_dns(log, cp):
+    """Retrieve the list of banned dns"""
+    banned_dns = []
+    try:
+        banned_dns_path = cp.get('auth', 'banned-dns')
+        banned_dns_file = os.path.normpath(
+            os.path.expandvars(banned_dns_path))
+    except ConfigParser.NoOptionError:
+        banned_dns_file = None
+    f = None
+    try:
+        with open(banned_dns_file, 'r') as f:
+            lines = f.readlines()
+            for line in lines:
+                if line.isspace() or line.strip().startswith('#'):
+                    continue
+                elif line.strip().startswith('/'):
+                    banned_dns.append(line.strip())
+                else:
+                    log.warning('DN in banned dns list is not in '
+                                'the correct format: %s', line)
+    finally:
```

Comment:
Now we have a context manager for the file, we can get rid of the `try...finally` wrapper (and the `f=None') as the file will close as soon as the handler is exited (whether normally or by exception).

Reply:
done